akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -306,6 +312,12 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
    * @param handler handler for each persisted `event`
    */
   final def persistAsync[A](event: A)(handler: A ⇒ Unit): Unit = {
+    if (insidePersistCallback) {
+      val ex = new UnsupportedOperationException("calling persistAsync from within the context of a persist/persistAsync callback is not supported!")
+      try onPersistFailure(ex, event, sequenceNr) finally context.stop(self)
```

Comment:
I don't think this is a persist failure, I think you can just throw the exception (similar to as if the handler throwed something)